hoomd/hpmc/UpdaterBoxMC.cc

```diff
@@ -450,7 +451,7 @@ void UpdaterBoxMC::update(unsigned int timestep)
         m_exec_conf->msg->notice(8) << "Box shear move performed at step " << timestep << std::endl;
         update_shear(timestep, rng);
         }
-    else if (move_type_select < m_Volume_weight + m_lnVolume_weight + m_Length_weight + m_Shear_weight + m_Aspect_weight)
+    else if (move_type_select <= m_Volume_weight + m_lnVolume_weight + m_Length_weight + m_Shear_weight + m_Aspect_weight)
```

Comment:
is there any consequence (i.e. bug resolved) due to this change?

Reply:
I don't think so. Assuming for the moment that `Saru` was capable of producing float values of exactly 1.0 (which is not clear given the bugs @mphoward has found), then there was a very small chance that `move_type_select` was exactly equal to the sum of all the weights and the warning would be produced and the box move skipped.

The new random generator does produce 1.0 outputs, so I changed this to `<=`. The code would probably be cleaner and more robust using integers to select move categories, but I'm not going to take the time to rewrite it now.